writer.go

```diff
@@ -264,20 +265,31 @@ func (w *Writer) Reopen(file string) error {
 		return err
 	}
 
-	// swap the unsafe pointer
-	oldfile := atomic.SwapPointer((*unsafe.Pointer)(unsafe.Pointer(&w.file)), unsafe.Pointer(newfile))
```

Comment:
reopen with atomic swap can be safe while writing concurrently.

Reply:
If I don't see the point of swapping the pointers if I have to reopen oldFile anyway. I'm not an expert so maybe you coul explain me why it is best. I can modify as you say it but I would still need to open oldfile again since I closed it in line 259.

Reply:
Ok, thanks for your contribution. I have to review the code workflow after leaving a long time.